Open Except quick action for git historical commit or stash views

alphathekiwi opened the pull request:
Closes #40851 

Release Notes:
- Added: Open excerpt quick action
- Improved: File finder and readonly files

Added:
- Historical file viewing from git commits with navigation to parent/modified [alt]+[enter]
- Keybind "OpenExcerpt" for opening files from commits/stashes with configurable actions
    - "open_from_cursor" - smart action that tries to nav to parent/modified (red/green) if cursor is in there
    - "open_head" - opens latest version of file
    - "open_finder" - applies the file path to the finder (useful when file was moved)
    - "open_parent" - opens before changes were made
    - "open_modified" - opens with the changes
- Settings UI controls for diff view Open Excerpt behavior see above
- Git commit view expanded with toolbar controls to configure quick actions

**Improved:**
- Permission prompt for editing readonly files with user confirmation dialog (offers saveas)
- File finder accepts query parameter when initialising enabling further integrations

TODO:
- Fix bugs in my code
- open_from_cursor doesn't actually work yet
- sometimes buffer load in as empty, ensure fallback is implemented

xipeng-jin commented:
Hi @alphathekiwi , FYI here [#42558](https://github.com/zed-industries/zed/pull/42558). If you like we can work together on here, and we can combine the PR together to make the reviewing easier.

alphathekiwi committed:
Added setting
Rough Draft
Changed file finder to allow query option

Changed file finder to allow query option to be passed in which sets its query
Added UI to control setting
